IECoreRI : Added support for render:sampleMotion option

danieldresser opened the pull request:
Hopefully this looks about right?  The more complex part of this functionality is in a Gaffer pull request that will go up shortly.

johnhaddon commented on src/IECoreRI/RendererImplementation.cpp:
```diff
@@ -278,6 +278,18 @@ void IECoreRI::RendererImplementation::setOption( const std::string &name, IECor
 	}
 
+
+	if( name == "render:sampleMotion" )
```
This shouldn't have the "render:" prefix, as I explained in previous emails. Options destined to be supported in all renderers (like this one) are unprefixed. Renderer specific options are prefixed.

johnhaddon commented on src/IECoreRI/RendererImplementation.cpp:
```diff
@@ -278,6 +278,18 @@ void IECoreRI::RendererImplementation::setOption( const std::string &name, IECor
+
+	if( name == "render:sampleMotion" )
+	{
+		if( m_options->readable().find( "ri:hider:sampleMotion" ) == m_options->readable().end() )
```
Shouldn't this be "ri:hider:samplemotion" (all lowercase)?

johnhaddon commented on src/IECoreRI/RendererImplementation.cpp:
```diff
@@ -278,6 +278,18 @@ void IECoreRI::RendererImplementation::setOption( const std::string &name, IECor
+	{
+		if( m_options->readable().find( "ri:hider:sampleMotion" ) == m_options->readable().end() )
+		{
+			// If we aren't specifying the specific RenderMan sample motion option, then drive it
```
Having the order of operations be important to the semantics seems potentially confusing and fragile. Maybe instead we could either say "sampleMotion" is just a straightforward alias to "ri:hider:samplemotion", or store both and resolve them (with the "ri:" one winning as you have here) in `worldEnd()`. I think I prefer the latter...

danieldresser commented:
Removed "render:" prefix from sampleMotion, and fixed case on ri:hider:samplemotion.

Only thing I haven't done is remove the early exit which keeps the raw "sampleMotion" out of m_options.  If I just removed this, I would start getting warnings about 'IECoreRI::RendererImplementation::setOption : Unknown option "sampleMotion".'.  Do I need to remove the early exit here and then add something to strip it out later, just so you can optionally use "sampleMotion" in getOption?  Or is this OK as is?

johnhaddon commented:
> Do I need to remove the early exit here and then add something to strip it out later, just so you can optionally use "sampleMotion" in getOption? Or is this OK as is?

OK, let's call it good.
